src/test_graph.py

```diff
@@ -1,109 +1,161 @@
+import pytest
+
 #test node constructor
 
-def test_node_init_value():
-    from graph import GNode
-    node1 = GNode(1)
-    assert node1.value == 1
+def test_node_init_value(gnode1):
+    assert gnode1.value == 1
 
 
-def test_node_init_edges():
-    from graph import GNode
-    node1 = GNode(1)
-    assert node1.edges == []
+def test_node_init_edges(gnode1):
+    assert gnode1.edges == set()
 
-def test_graph_init_list_of_nodes():
-    from graph import Graph
-    test_graph = Graph()
-    assert test_graph.list_of_nodes == []
+def test_graph_init_set_of_nodes(graph_empty):
```

Comment:
Really good to see that you are using these simple sanity-check type tests on empty data structures. They may seem redundant, but they are a great place to start.